### PR TITLE
OSDOCS-6642: DNS tracking Network Observability

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

Issue:

https://issues.redhat.com/browse/OSDOCS-7187

Link to docs preview:

Overview: https://63031--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/observing-network-traffic#network-observability-dns-overview_nw-observe-network-traffic

Working with DNS tracking: https://63031--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/observing-network-traffic#network-observability-dns-tracking_nw-observe-network-traffic
QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by skrthomas** [comment]
Hey @msherif1234 can you PTAL at this first draft for DNS tracking documentation?

**Comment by msherif1234** [comment]
overall LGTM I would to see high level description for what this feature is about 
```shell
DNS tracking with eBPF tracepoint hooks can serve various purposes:

- Network Monitoring: Gain insights into DNS queries and responses, helping network administrators identify unusual patterns, potential bottlenecks, or performance issues.

- Security Analysis: Detect suspicious DNS activities, such as domain name generation algorithms (DGA) used by malware, or identify unauthorized DNS resolutions that might indicate a security breach.

- Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps, tracking latency, and identifying misconfigurations.

Also feature is supported for IPv4 and IPv6 for both UDP and TCP protocols
```

**Review by jpinsonneau — APPROVED** [review]

File: modules/network-observability-dns-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+* Troubleshooting:: Debug DNS-related issues by tracing DNS resolution steps, tracking latency, and identifying misconfigurations.
+
+When DNS tracking is enabled, you can see the following metrics represented in a chart in the *Overview*. See the _Additional Resources_ of this section for more information about enabling and working with this view.
+
+* Top 5 average DNS latencies
+* Top 5 DNS response code
+* Top 5 DNS response code stacked with total
+
+This feature is supported for IPv4 and IPv6 for both UDP and TCP protocols.
```
Comment (msherif1234) [inline modules/network-observability-dns-overview.adoc:22]: we need to take off TCP for this release sorry :)

**Comment by msherif1234** [comment]
cc'd @memodi for his review and feedback

**Comment by skrthomas** [comment]
@memodi when you get a chance, can you PTAL at this doc for DNS tracking and lmk your feedback?

**Review by memodi — CHANGES_REQUESTED** [review]

**Comment by skrthomas** [comment]
Thanks @memodi I incorporated your feedback :)

File: modules/network-observability-dns-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+= Configuring DNS tracking
+You can configure graphical representation of DNS tracking of network flows in the *Overview* view. Using DNS tracking with eBPF tracepoint hooks can serve various purposes:
+
+* Network Monitoring: Gain insights into DNS queries and responses, helping network administrators identify unusual patterns, potential bottlenecks, or performance issues.
+
+* Security Analysis: Detect suspicious DNS activities, such as domain name generation algorithms (DGA) used by malware, or identify unauthorized DNS resolutions that might indicate a security breach.
+
+* Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps, tracking latency, and identifying misconfigurations.
+
+When DNS tracking is enabled, you can see the following metrics represented in a chart in the *Overview*. See the _Additional Resources_ of this section for more information about enabling and working with this view.
```
Comment (tmalove) [inline modules/network-observability-dns-overview.adoc:16]:
```suggestion
When DNS tracking is enabled, you can see the following metrics represented in a chart in the *Overview*. See the _Additional Resources_  section for more information about enabling and working with this view.
```
...or another option is " See Additional Resources in this section for more information about...".

File: modules/network-observability-dns-tracking.adoc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    ebpf:
+      features:
+      - DNSTracking           <1>                       
+      privileged: true        <2>
+----
+<1> You can set the `spec.agent.ebpf.features` parameter list to enable DNS tracking of each network flow in the web console.
+<2> Note that the `spec.agent.ebpf.privileged` specification value must be `true` for packet drop tracking to be enabled.
+
+
+. When you refresh the *Network Traffic* page, there are new choices to enable in the *Overview* and *Traffic Flow* views as well as new filters.
```
Comment (tmalove) [inline modules/network-observability-dns-tracking.adoc:41]:
```suggestion
. When you refresh the *Network Traffic* page, there are new choices to enable in the *Overview* and *Traffic Flow* views and new filters.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#a

File: modules/network-observability-dns-tracking.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+.. The *Overview* view has new DNS choices in the in *Manage panels* that you can select to see graphical visualizations and DNS metrics.
+.. In the *Traffic Flows* view new choices are available in *Manage columns* to add DNS columns to the table. 
+.. Next you can filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code* and see more information from the side panel.
+.. In *Query Options* verify that *Show duplicates* is selected in order to show duplicated DNS flows.
```
Comment (tmalove) [inline modules/network-observability-dns-tracking.adoc:45]:
```suggestion
.. In *Query Options* verify that *Show duplicates* is selected to show duplicated DNS flows.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#i

**Review by tmalove — APPROVED** [review]

**Comment by kalexand-rh** [comment]
Only versions 4.11+ are still in maintenance. I am removing labels from earlier versions from this PR.

File: modules/network-observability-dns-tracking.adoc (hunk 4 above)
Comment (skrthomas) [reply modules/network-observability-dns-tracking.adoc:45]: I'm actually going to end up removing this step. Thanks anyways @tmalove , sorry!!

**Comment by skrthomas** [comment]
4.11: https://github.com/openshift/openshift-docs/pull/65425

**Comment by skrthomas** [comment]
4.12: https://github.com/openshift/openshift-docs/pull/65427

**Pull request merged**